src/Psalm/Internal/Type/TemplateInferredTypeReplacer.php

```diff
@@ -442,15 +442,10 @@ private static function replaceConditional(
                         || $conditional_type->getId() !== 'float')
                 ) {
                     $matching_if_types[] = $candidate_atomic_type;
-                } elseif (!UnionTypeComparator::isContainedBy(
-                    $codebase,
-                    $conditional_type,
+                } elseif (null === Type::intersectUnionTypes(
```

Comment:
The logic is

When doing `A is B ? C : D`:
- If A is contained by B => It's C
- If B is contained by A => It's D
- Else, it's C|D

I think this logic is wrong and does not work well with string type like `literal-string` and `non-empty-string`.
it should be
- If A is contained by B => It's C
- If A&B = NEVER => It's D
- Else it's C|D